src/maxblocksize.cpp

```diff
@@ -0,0 +1,149 @@
+    }
+    else
+    {
+        uint64_t lowerFloor = nMaxBlockSize * 100 / 105;
+        lowerValue = (lowerValue < lowerFloor ? lowerFloor : lowerValue);
+        if (lowerValue < nMaxBlockSize)
+            nMaxBlockSize = lowerValue;
+    }
+
+    if (nMaxBlockSize != pindexLast->nMaxBlockSize)
+    {
+        LogPrintf("GetNextMaxBlockSize RETARGET\n");
+        LogPrintf("Before: %d\n", pindexLast->nMaxBlockSize);
+        LogPrintf("After:  %d\n", nMaxBlockSize);
+    }
+
+    return nMaxBlockSize;
+}
+
+static uint32_t FindVote(const std::string &coinbase)
+{
+    uint32_t eb_vote = 0;
+    std::vector<char> curr;
+    bool bip100vote = false;
+    bool started = false;
+
+    for (char s : coinbase)
+    {
+        if (s == '/')
+        {
+            started = true;
+            // End (or beginning) of a potential vote string.
+
+            if (curr.size() < 2) // Minimum vote string length is 2
+            {
+                bip100vote = false;
+                curr.clear();
+                continue;
+            }
+
+            if (std::string(begin(curr), end(curr)) == "BIP100")
+            {
+                bip100vote = true;
+                curr.clear();
+                continue;
+            }
+
+            // Look for a B vote.
+            if (bip100vote && curr[0] == 'B')
+            {
+                try
+                {
+                    return boost::lexical_cast<uint32_t>(std::string(begin(curr) + 1, end(curr)));
+                }
+                catch (const std::exception &e)
+                {
+                    LogPrintf("Invalid coinbase B-vote: %s\n", e.what());
+                }
+            }
+
+            // Look for a EB vote. Keep it, but continue to look for a BIP100/B vote.
+            if (!eb_vote && curr[0] == 'E' && curr[1] == 'B')
+            {
+                try
+                {
+                    eb_vote = boost::lexical_cast<uint32_t>(std::string(begin(curr) + 2, end(curr)));
+                }
+                catch (const std::exception &e)
+                {
+                    LogPrintf("Invalid coinbase EB-vote: %s\n", e.what());
+                }
+            }
+
+            bip100vote = false;
+            curr.clear();
+            continue;
+        }
+        else if (!started)
+            continue;
+        else
+            curr.push_back(s);
+    }
+    return eb_vote;
+}
+
+uint64_t GetMaxBlockSizeVote(const CScript &coinbase, int32_t nHeight)
+{
+    // Skip encoded height if found at start of coinbase
+    CScript expect = CScript() << nHeight;
+    int searchStart = coinbase.size() >= expect.size() && std::equal(expect.begin(), expect.end(), coinbase.begin()) ?
+                          expect.size() :
+                          0;
+
+    std::string s(coinbase.begin() + searchStart, coinbase.end());
+
+    if (s.length() < 5) // shortest vote is /EB1/
+        return 0;
+
+
+    return static_cast<uint64_t>(FindVote(s)) * 1000000;
```

Comment:
We specified only integral EB megabyte signals would count as votes, although in -bip100 mode, BU will publish (and enforce) EB with up to 6 decimal places (the need for which arises from the 5% change cap).

The reasons were to keep the semantics of B and EB the same, and to avoid decimal parsing logic.  The bip looks forward to a time when there will be several significant digits to the left of the decimal point.